lib/fastly/belongs_to_service_and_version.rb

```diff
@@ -23,8 +25,15 @@ def as_hash
       super.delete_if { |var| %w(service_id version).include?(var) }
     end
 
+    # URI escape (including spaces) the path and return it
+    def self.path_escape(path)
+      @uri_parser ||= URI::Parser.new
+      # the leading space in the escape character set is intentional
+      @uri_parser.escape(path, ' !*\'();:@&=+$,/?#[]')
```

Comment:
oh, that was intentional (and I should have mentioned it in the comment). Surprisingly, `\` isn't a reserved character but perhaps it's worth including in this list, especially for windows users? In our case, this came up because someone had named some custom VCL by the path it lived at in the repo so I can imagine it's a common thing to do.